pcdsdevices/interface.py

```diff
@@ -1693,20 +1693,22 @@ def __init__(self, *args,
         self._lightpath_ready = False
         self._retry_lightpath = False
         self._cached_state = None
-        self.input_branches = input_branches
-        self.output_branches = output_branches
+        self._md = None
 
         super().__init__(*args, **kwargs)
 
-        if self.input_branches and self.output_branches:
+        if input_branches and output_branches:
+            self.input_branches = input_branches
+            self.output_branches = output_branches
             self._init_summary_signal()
 
     def _init_summary_signal(self) -> None:
-        for sig in self.lightpath_cpts:
-            self.lightpath_summary.add_signal_by_attr_name(sig)
+        if self.lightpath_cpts:
+            for sig in self.lightpath_cpts:
+                self.lightpath_summary.add_signal_by_attr_name(sig)
 
-        self.lightpath_summary.subscribe(self._calc_cache_lightpath_state,
-                                         run=False)
+            self.lightpath_summary.subscribe(self._calc_cache_lightpath_state,
+                                             run=False)
```

Comment:
Do we need to guard against this running twice? Or does it not matter?

Reply:
That was meant to prevent the subscription if `lightpath_cpts` was empty, but that might be excessive.

Reply:
specifically: `_init_summary_signal` will be called twice if the branches are passed in init via happi, is that a problem?

Reply:
The scenario where happi is misconfigured to send these settings as kwargs to `__init__` could lead to double subscriptions and confusion. I think Zach is onto something - it's a good idea to add a guard here.

That's the only scenario that seems plausible to me. We won't have someone creating an instance on their own (with kwargs) and patching in `.md`.

Reply:
Oh I see what you mean.  I can add a check for that real quick.  The signals are stored as a dictionary and shouldn't be a problem, but duplicate subscriptions might be